### ISSUE TITLE
Implementing Node communication resolver ( LB traffic forwarding)

### ISSUE THREAD

**Issue by mogluu** [GluuFederation/oxCore#118]
![LoadBalancerforwarding](https://user-images.githubusercontent.com/47318409/54307805-71671b80-45c4-11e9-8859-aaeb97d9fde1.png)

**Comment by yurem** [comment]
My old design which I prepared before we added cache servers
![oxAuth horizontal scale](https://user-images.githubusercontent.com/6104200/54338411-e1b48200-4642-11e9-8622-64c68496f2da.png)
![oxAuth cluster process request](https://user-images.githubusercontent.com/6104200/54338416-e37e4580-4642-11e9-9b4d-5d62b57336b6.png)

**Comment by yuriyz** [comment]
oxauth is stateless, all data is on cache/persistence layer. LB knows traffic/load on each node. If some node goes down LB should switch traffic to active/green/alive node. Why do we need communication between nodes? On application layer it doesn’t matter to what node we switch thus no need track slaves (dependent nodes). The only thing we need is to track whether node is alive or not and we need LB which can do traffic switch.

**Comment by nynymike** [comment]
I agree that we don't want to introduce inter-oxauth communication. We already have a very complex deployment. This would transform oxauth into more like a database then a web tier component. I'd rather introduce communication to MQ component then better servers directly.

**Comment by mogluu** [comment]
Ok I guess we have to address maybe the most important LBs out there in a doc. The issue here is that if the LB fails to switch traffic the logs get loaded with associated errors. We also need to figure out how and which node is down when it goes down. In the case you you several nodes it can go with days without knowing.
  

  
Maybe a better implentation or idea ?   
  

  
  

  
  
  
  
  
>

**Comment by nynymike** [comment]
I think it would be better if there was a monitoring node that published a message in the cache that a certain node is out of commission

**Comment by yurem** [comment]
In my message is 2 years old design :). And the main idea was to find solution that not requires replication/cache or other cost resources.
